src/main/java/graphql/execution/ConditionalNodes.java

```diff
@@ -20,8 +20,11 @@
 
     public boolean shouldInclude(Map<String, Object> variables, List<Directive> directives) {
         boolean skip = getDirectiveResult(variables, directives, SkipDirective.getName(), false);
-        boolean include = getDirectiveResult(variables, directives, IncludeDirective.getName(), true);
-        return !skip && include;
+        if (skip) {
+            return false;
+        }
+
+        return getDirectiveResult(variables, directives, IncludeDirective.getName(), true);
```

Comment:
I think this is easier to understand as well.  I am not a fan of negative booleans